### ISSUE TITLE
[Bug] Add-DbaAgDatabase wont allow me to use already create backups

### ISSUE THREAD

**Issue by poweredbytones** [sqlcollaborative/dbatools#6714]
### Environmental information

PowerShell Version        : 5.1.17134.858
dbatools latest installed : 1.0.101
Culture of OS             : en-US

#### SQL Server:

```sql
Microsoft SQL Server 2017 (RTM-CU20) (KB4541283) - 14.0.3294.2 (X64) 
	Mar 13 2020 14:53:45 
	Copyright (C) 2017 Microsoft Corporation
	Developer Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor)

```

```sql
us_english
```

#### Host used

- [ X] powershell.exe
- [X ] ISE
- [ X] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [X ] Still buggy in powershell.exe

#### Errors Received

Add-DbaAgDatabase -SqlInstance $primary.Name -AvailabilityGroup $AG -Database $DBNAME -SeedingMode Automatic;
causes a back up to occur when we would like to utilize the already created backups.

#### Expected Behavior

I expected to be able to run using the backup already created not to run another costly backup

#### Actual Behavior

This command forces the run of a new backup.

**Comment by wsmelton** [comment]
@poweredbytones this is because you are using Automatic seeding and is a mechanism that Microsoft has that feature use in SQL Server. We have no control over that ability.

If you want to utilize the last backup for seeding you can `-UseLastBackup`: https://docs.dbatools.io/#Add-DbaAgDatabase

**Comment by andreasjordan** [comment]
This is the code that runs the backup:
https://github.com/sqlcollaborative/dbatools/blob/1f03b939f70f1961e7e9efe8b5f3ddcb0799d7b9/functions/Add-DbaAgDatabase.ps1#L171-L176

I think this code is not necessary.

**Comment by andreasjordan** [comment]
For manual seeding, there is a check for a full backup:

https://github.com/sqlcollaborative/dbatools/blob/1f03b939f70f1961e7e9efe8b5f3ddcb0799d7b9/functions/Add-DbaAgDatabase.ps1#L165-L169

I think this is to ensure that the active recovery model ist full. This would probably be a good idea for automatic seeding as well.

Let's have a look at the microsoft documentation:
https://docs.microsoft.com/en-us/sql/database-engine/availability-groups/windows/automatically-initialize-always-on-availability-group?view=sql-server-ver15

"Databases in an availability group must be in full recovery model. The database needs to have a current full backup and transaction log backup. These backup files are not used for automatic seeding, but they are required before including the database in an availability group."

So we should just check for having a full and log backup and do not take one.

**Comment by wsmelton** [comment]
The check at line 165 is to see if a full backup has ever been taken on the database. Checking for the year to be one is seeing if the last full backup date is turned as year 0001.

**Comment by wsmelton** [comment]
Code at 171 is just taking a null backup.

SQL Server will take an additional full backup when you use automatic seeding whether you have a full backup from 5 minutes ago it not.

**Comment by andreasjordan** [comment]
Let me quote myself about the null backup in line 171: "I think this code is not necessary."
And let me rephrase my question: What is the reason for this backup?
(If there's no reason for a line of code, I think it's a good idea to delete the line.)

**Comment by andreasjordan** [comment]
Hi @poweredbytones ,

let me copy some lines I posted on slack some hours ago, because that might help you:

I just had a look at my demos: It should work. Here is the key to success: First do a manual restore of the database (full and log backup) to the secondary instance and choose your custom path. Leave the database in restore mode (norecovery). Then you can use automatic seeding with Add-DbaAgDatabase.
Here is part of my demo:
```
$null = New-DbaDatabase -SqlInstance $SqlInstances[0] -Name TestDB2
$null = Backup-DbaDatabase -SqlInstance $SqlInstances[0] -Database TestDB2 -Path \\WIN10\SQLServerBackups -Type Full | Restore-DbaDatabase -SqlInstance $SqlInstances[1] -NoRecovery
$null = Backup-DbaDatabase -SqlInstance $SqlInstances[0] -Database TestDB2 -Path \\WIN10\SQLServerBackups -Type Log | Restore-DbaDatabase -SqlInstance $SqlInstances[1] -Continue -NoRecovery
Add-DbaAgDatabase -SqlInstance $SqlInstances[0] -AvailabilityGroup $AvailabilityGroupName -Database TestDB2 -Secondary $SqlInstances[1] -SeedingMode Automatic
```
Here without a custom path, but that should not make a difference. The key is that automatic seeding looks at the secondary instance and if it finds the database there in restoring mode is does not do the full backup but only transfers the needed log entries.
